controllers/install/binoculars_controller_unit_test.go

```diff
@@ -0,0 +1,77 @@
+package install
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/armadaproject/armada-operator/apis/install/v1alpha1"
+	installv1alpha1 "github.com/armadaproject/armada-operator/apis/install/v1alpha1"
+	"github.com/armadaproject/armada-operator/internal/k8sclient"
+)
+
+func TestBinocularsReconciler_Reconcile(t *testing.T) {
+	t.Parallel()
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	expectedNamespacedName := types.NamespacedName{Namespace: "default", Name: "binoculars"}
+	expectedBinoculars := v1alpha1.Binoculars{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Binoculars",
+			APIVersion: "install.armadaproject.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{Namespace: "default", Name: "binoculars"},
+		Spec: v1alpha1.BinocularsSpec{
+			Labels: nil,
+			Image: installv1alpha1.Image{
+				Repository: "testrepo",
+				Tag:        "1.0.0",
+			},
+			ApplicationConfig: runtime.RawExtension{},
+		},
+	}
+	mockK8sClient := k8sclient.NewMockClient(mockCtrl)
+	mockK8sClient.
+		EXPECT().
+		Get(gomock.Any(), expectedNamespacedName, gomock.AssignableToTypeOf(&v1alpha1.Binoculars{})).
+		Return(nil).
+		SetArg(2, expectedBinoculars)
+	mockK8sClient.
+		EXPECT().
+		Get(gomock.Any(), expectedNamespacedName, gomock.AssignableToTypeOf(&rbacv1.ClusterRole{})).
+		Return(errors.NewNotFound(schema.GroupResource{}, "binoculars"))
+	mockK8sClient.EXPECT().Create(gomock.Any(), gomock.AssignableToTypeOf(&rbacv1.ClusterRole{})).Return(nil)
+
+	//expectedSecret := corev1.Secret{
```

Comment:
Can we just go ahead and remove this, if it's not going to be used?

Reply:
I think it will be used.  Not sure why @dejanzele has it commented out in the executor but this is used to inject the application config into the PodTemplate.